src/Api/Monetization/Normalizer/ApiPackageNormalizer.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Apigee\Edge\Api\Monetization\Normalizer;
+
+use Apigee\Edge\Api\Monetization\Entity\ApiPackageInterface;
+use Apigee\Edge\Api\Monetization\NameConverter\ApiPackageNameConverter;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Symfony\Component\PropertyInfo\PropertyTypeExtractorInterface;
+use Symfony\Component\Serializer\Mapping\Factory\ClassMetadataFactoryInterface;
+use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+
+class ApiPackageNormalizer extends EntityNormalizer
+{
+    /**
+     * ApiPackageNormalizer constructor.
+     *
+     * @param null|\Symfony\Component\Serializer\Mapping\Factory\ClassMetadataFactoryInterface $classMetadataFactory
+     * @param null|\Symfony\Component\Serializer\NameConverter\NameConverterInterface $nameConverter
+     * @param null|\Symfony\Component\PropertyAccess\PropertyAccessorInterface $propertyAccessor
+     * @param null|\Symfony\Component\PropertyInfo\PropertyTypeExtractorInterface $propertyTypeExtractor
+     */
+    public function __construct(?ClassMetadataFactoryInterface $classMetadataFactory = null, ?NameConverterInterface $nameConverter = null, ?PropertyAccessorInterface $propertyAccessor = null, ?PropertyTypeExtractorInterface $propertyTypeExtractor = null)
+    {
+        $nameConverter = $nameConverter ?? new ApiPackageNameConverter();
+        parent::__construct($classMetadataFactory, $nameConverter, $propertyAccessor, $propertyTypeExtractor);
+    }
+
+    /**
+     * @inheritDoc
+     *
+     * @psalm-suppress InvalidReturnType Returning an object here is required
+     * for creating a valid Apigee Edge request.
+     */
+    public function normalize($object, $format = null, array $context = [])
+    {
+        $normalized = (array) parent::normalize($object, $format, $context);
+
+        // Do not send redundant API product information to Apigee Edge, the
+        // id of a referenced API product is enough.
+        foreach ($normalized['product'] as $id => $data) {
+            $normalized['product'][$id] = (object) ['id' => $data->id];
```

Comment:
I'm getting an  error because `data->id` is not set.  I think name is used for this object.

<img width="358" alt="image" src="https://user-images.githubusercontent.com/159693/47177700-00019600-d2ce-11e8-8b6b-9f89e74db3b6.png">

Reply:
What I can see is a Drupal (Mint?) API product entity (?), why API package normalizer would be called on that if everything configured properly? The line you commented gets called when a new API package gets created with a selected list of API products.

@Jaesin do you still have this issue?

Reply:
Hi @mxr576,
Sorry for not giving better context. I may have been using a management product as a package product. Would I have to create a management product then save it and fetch the monetizaton product in order to construct an API package?

Reply:
Yes, you should do that, if you check the annotation on the trait used by the ApiPackage entity you can see that: https://github.com/apigee/apigee-client-php/pull/18/files#diff-81c72e99743d59cd652977609a66ca0cR28.